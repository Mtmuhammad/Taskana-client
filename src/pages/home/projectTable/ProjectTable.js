import React from 'react'

const ProjectTable = ({projects}) => {
  return (
   <div className="mb-5 col-lg-6 d-flex align-items-stretch">
   <div data-testid="project-table" className="table-style card w-100">
     <div className="card-body">
       <div className="d-md-flex">
         <h3 className="card-title">Projects</h3>
       </div>
       <div>
         <div className="table-responsive mt-3">
           <table className="table stylish-table v-middle mb-0 no-wrap">
             <thead>
               <tr>
                 <th className="border-0 text-muted fw-normal">
                   Name
                 </th>
                 <th className="border-0 text-muted fw-normal">Date</th>
                 <th className="border-0 text-muted fw-normal">
                   Status
                 </th>
                 <th className="border-0 text-muted fw-normal">Deadline</th>
               </tr>
             </thead>
             <tbody>
              { projects?.map((project, index) => (<tr key={index} className="h5">
                 <td>
                   <p className="font-weight-medium mb-0">{project?.name}</p>
                 </td>
                 <td>{project?.date}</td>
                 <td>
                   <span className="badge bg-success px-2 py-1">
                     {project?.status}
                   </span>
                 </td>
                 <td>{project?.deadline}</td>
               </tr>))}
               
             </tbody>
           </table>
         </div>
       </div>
     </div>
   </div>
 </div>
  )
}

export default ProjectTable