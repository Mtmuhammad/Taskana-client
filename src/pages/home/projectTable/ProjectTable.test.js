import React from "react";
import { render, screen } from "@testing-library/react";
import ProjectTable from "./ProjectTable";
import { BrowserRouter } from "react-router-dom";

describe("smoke and snapshot tests", () => {
  test("should render without crashing", () => {
    render(<ProjectTable />, { wrapper: BrowserRouter });
  });

  test("should match snapshot", () => {
    const { asFragment } = render(<ProjectTable />, { wrapper: BrowserRouter });
    expect(asFragment()).toMatchSnapshot();
  });
});

describe("project-table", () => {
  test("should render a project table", () => {
    render(<ProjectTable />, { wrapper: BrowserRouter });
    const table = screen.getByTestId("project-table");
    expect(table).toBeInTheDocument();
    expect(table).toHaveClass("table-style card");
  });

  test("should render table heading", () => {
    render(<ProjectTable />, { wrapper: BrowserRouter });
    const heading = screen.getByText("Projects");
    expect(heading).toBeInTheDocument();
    expect(heading).toHaveClass("card-title");
  });

  test("should render table headers", () => {
    render(<ProjectTable />, { wrapper: BrowserRouter });
    const headers = screen.getAllByRole("columnheader");
    expect(headers.length).toBe(4);
  });
});
