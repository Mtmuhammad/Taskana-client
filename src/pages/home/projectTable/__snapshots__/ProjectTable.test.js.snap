// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`smoke and snapshot tests should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="mb-5 col-lg-6 d-flex align-items-stretch"
  >
    <div
      class="table-style card w-100"
      data-testid="project-table"
    >
      <div
        class="card-body"
      >
        <div
          class="d-md-flex"
        >
          <h3
            class="card-title"
          >
            Projects
          </h3>
        </div>
        <div>
          <div
            class="table-responsive mt-3"
          >
            <table
              class="table stylish-table v-middle mb-0 no-wrap"
            >
              <thead>
                <tr>
                  <th
                    class="border-0 text-muted fw-normal"
                  >
                    Name
                  </th>
                  <th
                    class="border-0 text-muted fw-normal"
                  >
                    Date
                  </th>
                  <th
                    class="border-0 text-muted fw-normal"
                  >
                    Status
                  </th>
                  <th
                    class="border-0 text-muted fw-normal"
                  >
                    Deadline
                  </th>
                </tr>
              </thead>
              <tbody />
            </table>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
